### PR TITLE
Fixing JSON schema conformance

### PR DESCRIPTION
**Pull request by wpbrown**
Fixing JSON schema spec conformance problems. These problems cause exceptions when trying to load the ARM schema in Newtonsoft Json.NET Schema.

### CODE REVIEW

File: schemas/2015-01-01/deploymentTemplate.json [hunk 1]
```diff
@@ -904,25 +904,25 @@
...
-                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#resourceDefinitions/pricings"},
-                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#resourceDefinitions/securityContacts"},
-                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#resourceDefinitions/workspaceSettings"},
-                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#resourceDefinitions/autoProvisioningSettings"},
-                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#resourceDefinitions/advancedThreatProtectionSettings"},
-                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#resourceDefinitions/informationProtectionPolicies"},
+                  { "$ref": "https://schema.management.azure.com/schemas/2015-03-01-preview/Microsoft.HDInsight.json#/resourceDefinitions/clusters" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >/ [](start = 117, length = 1)

I assume this is the focus of the change?

File: schemas/2015-01-01/deploymentTemplate.json [hunk 2]
```diff
@@ -904,25 +904,25 @@
...
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-08-01-preview/Microsoft.Security.json#/resourceDefinitions/informationProtectionPolicies"},
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#/resourceDefinitions/pricings"},
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#/resourceDefinitions/securityContacts"},
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#/resourceDefinitions/workspaceSettings"},
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#/resourceDefinitions/autoProvisioningSettings"},
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#/resourceDefinitions/advancedThreatProtectionSettings"},
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-06-01/Microsoft.Security.json#/resourceDefinitions/informationProtectionPolicies"},
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >resourceDefinitions [](start = 109, length = 19)

do we need to update 2014-04-01-preview/deploymentTemplate.json as well?

**Comment by Tiano2017** [comment]
@wpbrown Thanks for making the change. there are some conflicts with origin/master. I also left two comment. Can you take a look?

File: schemas/2015-01-01/deploymentTemplate.json (hunk 1 above)
Comment (wpbrown) [reply schemas/2015-01-01/deploymentTemplate.json]: Yes the only change in each line is the new `/` after the `#`. Without it, a reference to an "id" is implied.

File: schemas/2015-01-01/deploymentTemplate.json (hunk 2 above)
Comment (wpbrown) [reply schemas/2015-01-01/deploymentTemplate.json]: I wasn't familiar with that one, but it would seem so. I'll update it with the same change.

**Comment by wpbrown** [comment]
@Tiano2017 I merged master and updated 2014-04-01-preview as well.

**Pull request merged**